From patchwork Fri Dec 18 02:47:39 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Stephen Rothwell <sfr@canb.auug.org.au>
X-Patchwork-Id: 11981079
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-10.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_PATCH,
	MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS autolearn=unavailable
	autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 3C039C2BBCD
	for <kvm@archiver.kernel.org>; Fri, 18 Dec 2020 02:48:49 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 07A6023A7B
	for <kvm@archiver.kernel.org>; Fri, 18 Dec 2020 02:48:49 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1732463AbgLRCsd (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 17 Dec 2020 21:48:33 -0500
Received: from ozlabs.org ([203.11.71.1]:57001 "EHLO ozlabs.org"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1726851AbgLRCsc (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 17 Dec 2020 21:48:32 -0500
Received: from authenticated.ozlabs.org (localhost [127.0.0.1])
        (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
         key-exchange ECDHE (P-256) server-signature RSA-PSS (4096 bits)
 server-digest SHA256)
        (No client certificate requested)
        by mail.ozlabs.org (Postfix) with ESMTPSA id 4CxtZZ1Hvjz9sWR;
        Fri, 18 Dec 2020 13:47:50 +1100 (AEDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=canb.auug.org.au;
        s=201702; t=1608259670;
        bh=9kLtd9vZRVb0Bzbu/us9QjmM3pXtlm84GlIUkd9FX7o=;
        h=Date:From:To:Cc:Subject:From;
        b=uZfdmiUHzYNu+cxFXGM/P0qEivh5ZpvXsloSwyFWIMSbpIhK6zxaX8EZieGJDEksM
         eZSsQljpZRXskqU6qgGot0dRN+rZSFhNzosAMbGd9x2Dc7BxcRxGfZoNe8Gj61Twwc
         jXBKc14aEZmSJor0sPWU+58J8/G5YTSGQLnplmcO7ASQs42cgjgqbuRPRFLV8SYHmb
         3O73EwWUYWRT4KJ69RbDHKE86SoObvbIQ6EkjFKcwzB3XxfECZeWG9ECiNwpX4ViCj
         HWwEFfs0UofEXtaGj5dlfAGhvEd8qjrXvat1GmGETbraYP7IU7P5a6CBufanmBKGaR
         kv4mrrNk5aDoA==
Date: Fri, 18 Dec 2020 13:47:39 +1100
From: Stephen Rothwell <sfr@canb.auug.org.au>
To: Paolo Bonzini <pbonzini@redhat.com>, KVM <kvm@vger.kernel.org>,
        Catalin Marinas <catalin.marinas@arm.com>,
        Will Deacon <will@kernel.org>
Cc: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>,
        Linux Next Mailing List <linux-next@vger.kernel.org>,
        Marc Zyngier <maz@kernel.org>
Subject: linux-next: manual merge of the kvm tree with the arm64-fixes tree
Message-ID: <20201218134739.45973671@canb.auug.org.au>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Hi all,

Today's linux-next merge of the kvm tree got a conflict in:

  arch/arm64/include/asm/kvm_asm.h

between commit:

  9fd339a45be5 ("arm64: Work around broken GCC 4.9 handling of "S" constraint")

from the arm64-fixes tree and commit:

  b881cdce77b4 ("KVM: arm64: Allocate hyp vectors statically")

from the kvm tree.

I fixed it up (see below) and can carry the fix as necessary. This
is now fixed as far as linux-next is concerned, but any non trivial
conflicts should be mentioned to your upstream maintainer when your tree
is submitted for merging.  You may also want to consider cooperating
with the maintainer of the conflicting tree to minimise any particularly
complex conflicts.

diff --cc arch/arm64/include/asm/kvm_asm.h
index 8e5fa28b78c2,7ccf770c53d9..000000000000
--- a/arch/arm64/include/asm/kvm_asm.h
